### PR TITLE
Add support for change events on `<select>` elements

### LINKED ISSUE #8939: amp-bind: Support <select>
We should support setting state vars from drop-down menus. One way to do this is add support for `<select>`'s change event, as we currently do for `<input>`:
```
<select on="change:AMP.setState({event.value})">
  <option value="foo">foo</option>
  <option value="bar">bar</option>
</select>
```
/to @kmh287

### PR DESCRIPTION
**Pull request by kmh287**
Mostly just expands on the work done to support `input` elements. 

Fixes #8939

### CODE REVIEW

File: src/service/action-impl.js [hunk 1]
```diff
@@ -216,6 +216,9 @@ export class ActionService {
         });
         event.detail = detail;
       }
+    } else if (event.target.tagName.toLowerCase() === 'select') {
```
Comment (choumx) [inline src/service/action-impl.js]: Recommend moving LHS to a local `tagName` var and use a switch statement.

Comment (kmh287) [inline src/service/action-impl.js]: good idea

File: src/service/action-impl.js [hunk 2]
```diff
@@ -199,23 +199,30 @@ export class ActionService {
...
-          const value = target[field];
-          if (expectedType === 'number') {
-            detail[field] = Number(value);
-          } else if (expectedType === 'boolean') {
-            detail[field] = !!value;
-          } else {
-            detail[field] = String(value);
-          }
-        });
+    const tagName = event.target.tagName.toLowerCase();
```
Comment (choumx) [inline src/service/action-impl.js]: Nit: `target` instead of `event.target`.

File: src/service/action-impl.js [hunk 3]
```diff
@@ -199,23 +199,30 @@ export class ActionService {
...
+              detail[field] = !!value;
+            } else {
+              detail[field] = String(value);
+            }
+          });
+          event.detail = detail;
+        }
+        break;
+      case 'select':
+        detail['value'] = target['value'];
```
Comment (choumx) [inline src/service/action-impl.js]: `detail.value = target.value;`?

File: src/service/action-impl.js [hunk 4]
```diff
@@ -199,23 +199,30 @@ export class ActionService {
   addChangeDetails_(event) {
     const detail = {};
```
Comment (choumx) [inline src/service/action-impl.js]: `const detail = map();`

File: src/service/action-impl.js [hunk 5]
```diff
@@ -199,23 +199,30 @@ export class ActionService {
...
+            const value = target[field];
+            if (expectedType === 'number') {
+              detail[field] = Number(value);
+            } else if (expectedType === 'boolean') {
+              detail[field] = !!value;
+            } else {
+              detail[field] = String(value);
+            }
+          });
+          event.detail = detail;
```
Comment (choumx) [inline src/service/action-impl.js]: Probably can take this out of the `switch` to reduce dupe.

File: src/service/action-impl.js (hunk 4 above)
Comment (kmh287) [inline src/service/action-impl.js]: done

File: src/service/action-impl.js (hunk 2 above)
Comment (kmh287) [inline src/service/action-impl.js]: done

File: src/service/action-impl.js (hunk 5 above)
Comment (kmh287) [inline src/service/action-impl.js]: ok, but we should only set this if `detail` isn't empty

File: src/service/action-impl.js (hunk 3 above)
Comment (kmh287) [inline src/service/action-impl.js]: done

**Pull request merged**